src/proposal.mligo

```diff
@@ -363,50 +371,79 @@ let handle_proposal_is_over
     let cons = fun (l, e : operation list * operation) -> e :: l in
     let ops = List.fold cons ops new_ops in
     let store = delete_proposal (start_date, proposal_key, store) in
-    (ops, store, counter)
-  else (ops, store, counter)
+    (ops, store, true)
+  else (ops, store, false)
 
-// Flush all proposals that passed their voting period.
-let flush(n, config, store : nat * config * storage): return =
-  let store =
-    if n = 0n
-      then
-        (failwith("BAD_ENTRYPOINT_PARAMETER") : storage)
-      else store
-    in
+// Elem function for list of `proposal_key`
+[@inline]
+let elem_proposal_key_list (l, ls : proposal_key * proposal_key list) : bool =
+  let check_equal (acc, e : bool * proposal_key) =
+        if e = l then true else false
+  in List.fold check_equal ls false
 
-  let counter : counter =
-    { current = 0n
-    ; total = n
-    } in
-  let flush_one
-      (acc, e: (operation list * storage * counter) * (timestamp * proposal_key)) =
-        let (ops, store, counter) = acc in
-        let (start_date, proposal_key) = e in
-        handle_proposal_is_over (config, start_date, proposal_key, store, ops, counter)
-      in
-  let (ops, store, counter) =
-    Set.fold flush_one store.proposal_key_list_sort_by_date (([] : operation list), store, counter) in
-  (ops, store)
-
-// Removes an accepted and finished proposal by key.
-let drop_proposal (proposal_key, config, store : proposal_key * config * storage): return =
-  let store = authorize_admin store in
+// Flush all proposals that passed their voting period.
+let flush(param, config, store : flush_param * config * storage): return =
+  match param with
+    | Flush_amount n ->
+        let store =
+          if n = 0n
+            then
+              (failwith("BAD_ENTRYPOINT_PARAMETER") : storage)
+            else store
+          in
+
+        let counter : counter =
+          { current = 0n
+          ; total = n
+          } in
+        let flush_one
+            (acc, e: (operation list * storage * counter) * (timestamp * proposal_key)) =
+              let (ops, store, counter) = acc in
+              let (start_date, proposal_key) = e in
+              if (counter.current < counter.total)
```

Comment:
Wonder if we can just use an integer counter, and just check if it greater than zero at every itereration, so that we don't have to pass `total` into this function and also can save the `counter` type.